tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -328,26 +328,33 @@ Internal::FriendInfo TTreeProcessorMT::GetFriendInfo(TTree &tree)
 
    for (auto fr : *friends) {
       const auto frTree = static_cast<TFriendElement *>(fr)->GetTree();
+      const bool isChain = frTree->IsA() == TChain::Class();
 
-      // Check if friend tree has an alias
-      const auto realName = frTree->GetName();
-      const auto alias = tree.GetFriendAlias(frTree);
-      if (alias) {
-         friendNames.emplace_back(std::make_pair(realName, std::string(alias)));
-      } else {
-         friendNames.emplace_back(std::make_pair(realName, ""));
-      }
-
-      // Store the file names of the friend tree
       friendFileNames.emplace_back();
       auto &fileNames = friendFileNames.back();
-      const bool isChain = tree.IsA() == TChain::Class();
+
       if (isChain) {
-         const auto frChain = static_cast<TChain *>(frTree);
-         for (auto f : *(frChain->GetListOfFiles())) {
+         const auto chainFiles = static_cast<TChain*>(frTree)->GetListOfFiles();
+         // TODO: This does not yet take into account different names for each tree in the chain.
```

Comment:
if you are not going to fix that case in this PR please open a jira ticket, I'll take a look when I'm back

Reply:
..just saw the ticket :smile: then this TODO is not needed

Reply:
Removed it :)